var/spack/repos/builtin/packages/mgcfd-op2/package.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class MgcfdOp2(MakefilePackage):
+    """Package for the OP2 port of MGCFD: A 3D unstructured multigrid,
+    finite-volume computational fluid dynamics (CFD) mini-app for inviscid-flow."""
+
+    # NOTE: This package is new and has been tested on a limited set of use cases:
+    # Graviton 2, Graviton 3:
+    #       Compilers: GCC 12.1.0, Arm 22.0.1, NVHPC 22.3
+    # ThunderX2 (Cray):
+    #       Compilers: GCC 10.3.0, CCE 11.0.4
+
+    homepage = "https://github.com/warwick-hpsc/MG-CFD-app-OP2"
+    git = "https://github.com/warwick-hpsc/MG-CFD-app-OP2.git"
+
+    maintainers = ["tomdeakin", "gihanmudalige", "bozbez"]
+
+    version("v1.0.0-rc1")
+
+    variant("mpi", default=False, description="Enable MPI support")
+
+    depends_on("gmake@4.3")
+    # KaHIP is a new MGCFD-OP2 dependency and
+    # NVHPC builds require the latest develop branch at time of writing (Sept 22)
+    depends_on("kahip@develop+metis", when="+mpi")
+    depends_on("op2-dsl+mpi", when="+mpi")
+    depends_on("op2-dsl~mpi", when="~mpi")
+
+    def edit(self, spec, prefix):
+        compiler_map = {
+            "gcc": "gnu",
+            "arm": "clang",
+            "cce": "cray",
+            "nvhpc": "pgi",
+        }
+
+        if self.spec.compiler.name in compiler_map:
+            env["COMPILER"] = compiler_map[self.spec.compiler.name]
+        else:
+            env["COMPILER"] = self.spec.compiler.name
```

Comment:
This is better suited for `setup_build_environment`

Reply:
Done - thanks